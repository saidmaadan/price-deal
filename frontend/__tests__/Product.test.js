import ProductComponent from '../components/Product';
import { shallow, mount} from 'enzyme';
import toJSON from 'enzyme-to-json';

const fakeProduct = {
    id: 'PD89G',
    title: 'Brown leather bag',
    price: 8000,
    description: 'New arrival women bag!',
    image: 'bag.jpg',
    largeImage: 'largebag.jpg',
};

describe('<Product/>', () => {
    it('renders and matches the snapshot', () => {
        const wrapper = shallow( < ProductComponent product={fakeProduct} />);

        expect(toJSON(wrapper)).toMatchSnapshot();
    });
    
    // it('renders the image properly', () => {
    //     const wrapper = shallow(<ProductComponent product={fakeProduct} />);
    //     const img = wrapper.find('img');
    //     expect(img.props().src).toBe(fakeProduct.image);
    //     expect(img.props().alt).toBe(fakeProduct.title);
    // });
    // it('renders the pricetag and title', () => {
    //     const wrapper = shallow(<ProductComponent product={fakeProduct} />);
    //     const PriceTag = wrapper.find('PriceTag');
    //     expect(PriceTag.children().text()).toBe('$80');
    //     expect(wrapper.find('Title a').text()).toBe(fakeProduct.title);
    // });
    // it('renders out the buttons properly', () => {
    //     const wrapper = shallow(<ProductComponent product={fakeProduct} />);
    //     const buttonList = wrapper.find('.buttonList');
    //     expect(buttonList.children()).toHaveLength(3);
    //     expect(buttonList.find('Link')).toHaveLength(1);
    //     expect(buttonList.find('DeleteProduct').exists()).toBe(true);
    // });
});