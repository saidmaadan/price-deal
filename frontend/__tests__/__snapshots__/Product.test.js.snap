// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Product/> renders and matches the snapshot 1`] = `
<ProductStyles__Product>
  <img
    alt="Brown leather bag"
    src="bag.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/product",
          "query": Object {
            "id": "PD89G",
          },
        }
      }
    >
      <a>
        Brown leather bag
      </a>
    </Link>
  </Title>
  <PriceTag>
    $80
  </PriceTag>
  <p>
    New arrival women bag!
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "PD89G",
          },
        }
      }
    >
      <a>
        Edit ✏️
      </a>
    </Link>
    <addToCart
      id="PD89G"
    />
    <DeleteProduct
      id="PD89G"
    >
      Delete This Product
    </DeleteProduct>
  </div>
</ProductStyles__Product>
`;
